### DISCUSSION TITLE
How to have project specific settings?

### DISCUSSION THREAD

**Discussion by stherold** [jdx/mise#1424] [Q&A] [answered]
I need to specify settings for our CI. Therefore I have to specify prohject specific settings, e.g. under `<project>/.config/mise/settings.toml`. I did not find any environment variable or parameter to specify an alternate settings file or something similar.

Is there a way to do that?

**Comment by jdx** [comment, marked as answer]
`MISE_SETTINGS_FILE`?

**Reply by stherold** [reply]
Oh it is not on this web-site: https://mise.jdx.dev/configuration.html#settings-file-config-mise-settings-toml

**Reply by jdx** [reply]
I'll fix that

**Reply by stherold** [reply]
Thanks that works great 👍

**Reply by alexanderwe** [reply]
@jdx Is it possible to set this directly in `mise.toml`  `[env]` section and let it point to a file relative to `.mise.toml` ? I tried it and it seems to not work when I am in a child directory of the root of my project.

The reason why I am asking is basically how to have project based settings without the need to always specify an environment variable in any directory below the project root directory and when `cd` out of the project, the global settings apply.

**Reply by jdx** [reply]
No the settings affect config loading so I think that's impossible

**Reply by alexanderwe** [reply]
Thanks a lot for the fast answer :) So this means if I for example want to have experimental features only enabled for one project but disabled for another, this is not possible ?  Only by always specifying the path to the settings with an ENV variable and not by having a file within the projects directory ? Because I have to say, as a beginner I liked to have the settings within the `mise.toml` file 

Sorry if the questions sound stupid, but just getting started with mise

**Reply by jdx** [reply]
for now, but I have plans to fix that and #1439 is the first step to pave the way

**Reply by alexanderwe** [reply]
That's great to hear !

**Reply by stherold** [reply]
To make it work in the meantime, my project specific **settings** file is under `<project>/.config/mise/settings.toml` and I set `MISE_SETTINGS_FILE=".config/mise/settings.toml"` in my .envrc file which is automatically loaded by [direnv](https://direnv.net/) when I cd into the directory. Having a bit of diversification in the tools is not always a bad idea 😅

**Reply by jdx** [reply]
direnv/mise work fine together, except sometimes where you start using direnv to manage PATH env vars. That I would discourage though I have worked pretty hard to at least _attempt_ to get it to work correctly.

**Comment by stherold** [comment]
Hey there 👋 

though it is not impossible, I recognized some challenges to get project specific settings to work (read other comments here). I just wonder what the motivation was to separate settings from the default config file?

Cheers 🥂
